### DISCUSSION TITLE
How to leave the browser open

### DISCUSSION THREAD

**Discussion by Yrr0r** [PDConSec/vsc-print#225] [Q&A] [answered]
Everytime I hit the print button this extension immediately opens the browser and a print dialog is immediately poped up for print. And if I hit 'cancel', the document is closed and gone. I found out this is behaviour is caused from a small JavaScript snippet injected into the head section. 

Can I request a setting that disables this behaviour? I wanted to view the document in the browser before it gets to print. And I also wonder if there's a possibility to just save the generated files so they can be viewed from any browser.

**Comment by PeterWone** [comment, marked as answer]
Uncheck the Print and Close setting.

![image](https://user-images.githubusercontent.com/5498936/224170382-1f0511a8-9704-421a-b4f4-73f08b89c840.png)

**Comment by PeterWone** [comment]
The extension doesn't directly support saving to a file because multiple files are involved. At a minimum, an HTML file and a CSS file. Depending on what you print there may also be image files or more CSS files. 

However, some browsers do support saving it all into a single file. Or at least they used to - I just looked for it and couldn't find it. But there is this https://microsoftedge.microsoft.com/addons/detail/save-as-mhtml/pjjgdpchkljpkhogkniodkimiaoeebmf and Edge still seems to be able to use an MHTML file. There seems to be one for Chrome too https://chrome.google.com/webstore/detail/save-as-mhtml/ahgakckdonjmnpnegjcamhagackmjpei which supports MHTML.

**Comment by RobCranfill** [comment]
I believe the first part of this request is already implemented: the "Print and Close" option in the extension's settings. If you turn this off, the printable code window opens, but the print dialog does not automatically come up.

**Comment by PeterWone** [comment]
@Yrr0r Rob is correct - this is what I meant by the screen snap. So I'm going to close this issue. I appreciate your interest and input. If you think of anything else don't hesitate to raise another issue.
